R/writeModel.R

```diff
@@ -25,62 +23,71 @@
 ##' @param con passed to \code{\link{writeLines}} which actually writes the
 ##' model file
 ##' @param digits number of significant digits used for \pkg{WinBUGS}
 ##' input, see \code{\link{formatC}}
-##' @return Nothing, but as a side effect, the model file is written
 ##'
 ##' @export
-##' @author original idea by Jouni Kerman, modified by Uwe Ligges, DSB removed S
-##' Plus part
-writeModel <- function(model, con = "model.bug", digits = 5)
-{
-    if (is.R()){
-        model.text <- c("model", replaceScientificNotationR(body(model),
-                                                            digits = digits))
-                                        # "[\+\-]?\d*\.?[Ee]?[\+\-]?\d*"
-    } else {
-        ## In S-PLUS the source code of a function can be obtained with
-        ## as.character(function_name).
-        ## This omits the "function_name <- function()" piece
-        model.text <- paste("model", as.character(model))
-    }
-    model.text <- gsub("%_%", "", model.text)
-    writeLines(model.text, con = con)
+##' @author original idea by Jouni Kerman, modified by Uwe Ligges,
+##'  DSB removed S-Plus part, WW modified the implementation.
+writeModel <- function(model, con = "model.bug", digits = 5) { # nolintr
+  if (is.R()) {
+    model_text <- c("model", h_replace_scientific(model, digits = digits))
+  } else {
+    # In S-PLUS the body of a function can be obtained with
+    # as.character(function_name).
+    model_text <- paste("model", as.character(model))
+  }
+  model_text <- gsub("%_%", "", model_text)
+  writeLines(model_text, con = con)
 }
 
-##' @keywords internal
-replaceScientificNotationR <- function(bmodel, digits = 5){
-    env <- new.env()
-    assign("rSNRidCounter", 0, envir=env)
-    replaceID <- function(bmodel, env, digits = 5){
-        for(i in seq_along(bmodel)){
-            if(length(bmodel[[i]]) == 1){
-                if(as.character(bmodel[[i]]) %in% c(":", "[", "[[")) return(bmodel)
-                if((typeof(bmodel[[i]]) %in% c("double", "integer")) && ((abs(bmodel[[i]]) < 1e-3) || (abs(bmodel[[i]]) > 1e+4))){
-                    counter <- get("rSNRidCounter", envir=env) + 1
-                    assign("rSNRidCounter", counter, envir=env)
-                    id <- paste("rSNRid", counter, sep="")
-                    assign(id, formatC(bmodel[[i]], digits=digits, format="E"), envir=env)
-                    bmodel[[i]] <- id
-                }
-            } else {
-                bmodel[[i]] <- replaceID(bmodel[[i]], env, digits = digits)
-            }
-        }
-        bmodel
-    }
-    bmodel <- deparse(replaceID(bmodel, env, digits = digits), control = NULL)
-    for(i in ls(env)){
-        bmodel <- gsub(paste('"', i, '"', sep=''), get(i, envir=env), bmodel, fixed=TRUE)
+h_replace_scientific <- function(model, digits = 5) {
+  assert_function(model)
+  assert_int(digits)
+
+  bmodel <- h_bapply(
+    body(model),
+    h_format_number,
+    classes = c("integer", "numeric"),
+    digits
```

Comment:
since you specify the prefix and suffix again in line 63, I would also specify them here, otherwise difficult to keep the code in sync

Reply:
Sure, so we just ignore the default values of `pref`, `suf` in `h_format_number`?